tests/utils.py

```diff
@@ -50,23 +51,32 @@ def __init__(
         self.server.socket = sslcontext.wrap_socket(self.server.socket, server_side=True)
 
 
+def _n_to_port(n: int) -> int:
+    """Provided with an integer 7..64, returns a port number 7777 to 64444"""
+    prefix: str = str(n)
+    last_digit: str = prefix[-1]
+    port: str = prefix + (last_digit * 3)
+    return int(port)
+
+
 _T_co = TypeVar('_T_co', bound=HttpServerThread, covariant=True)
 
 
 def create_server(
     server_thread_class: type[_T_co],
 ) -> Callable[[type[BaseRequestHandler]], AbstractContextManager[_T_co]]:
-    port_counter = AtomicInteger(7777)
+    counter = AtomicInteger(7)  # start from port 7777
```

Comment:
No need for an atomic counter. Just generate the set of ports and use `itertools.cycle` to cycle over. If a port is busy, the file will be locked (can you quickly check that the file is removed once it's unlocked actually?)

Reply:
`filelock`'s documentation confirms that it _does not_ remove the lockfiles for us: https://github.com/tox-dev/filelock/blob/141f5d8c21be2830a9d93ad4ad822acf4b0f8a12/src/filelock/_api.py#L275-L276

> [!NOTE]
> Releases the file lock. Please note, that the lock is only completely released, if the lock counter is 0. Also note, that the lock file itself is not automatically deleted.

Reply:
Ok, it's actually fine now that I think about. Namely, if you exhaust the current set of ports, you'll be able to use another port only after the lock (for this port) has been completely released.

Reply:
So, I would actually prefer not to use an atomic counter and generate the set of ports beforehand. Otherwise you'll end up with incremental port numbers:

```
7 -> 7777
8 -> 8888
9 -> 9999 
10 -> 10_000
11 -> 11_111
...
19 -> 19_999
...
```

You might exhaust the valid port ranges. And I prefer not to add an additional dependency. So fixed set of port numbers + (itertools.cycle + multiprocessing reentrant lock) + filelock is fine.